### ISSUE TITLE
Cannnot delete polygon label

### ISSUE THREAD

**Issue by wosiu** [wkentaro/labelme#1015]
I chosen a polygon on the list and git "Delete" button and I got this:
![image](https://user-images.githubusercontent.com/4964388/167964458-8070f454-62b9-4395-87c9-55660c053b32.png)

The label is still there after clicking "Yes".

**Comment by weiyixing** [comment]
yes .I have the same problem. I also open 'Save Automatically' in File ,but when I delete a polygon label ,then I  turn to next image . when I return  the image I deleted the polygon ,the polygon still exist . my labelme version is 4.6.0

**Comment by wosiu** [comment]
I also noticed, that it sometimes eventually gets deleted. But there is a huge lag in UI. The delayed action happens after I do some other action in UI.

**Comment by weiyixing** [comment]
I  know  how to deal with my question. my error is that ,when I make my  json label , I fill in  the json['imagePath']  the absolute path .but  the right is the name of image .
error example : "imagePath": "F:\\wyx\\stoneData\\test_data\\\u6d4b\u8bd5\u96c6\\5-10\\4_280001.jpg"
right example :'4_280001.jpg'
so may be your json format error ?

**Comment by wkentaro** [comment]
Thanks for the report. I'll look at this.

**Comment by NiklasDHahn** [comment]
@wkentaro are there any news on this bug? Apart from that, really love the program!

**Comment by wkentaro** [comment]
I tried to reproduce and couldn't, so please let me know a way to reproduce.

**Comment by NiklasDHahn** [comment]
Hey @wkentaro I think in this case it is user error, at least from my side. I did not see the "Edit Polygons" button which has to be clicked in order to delete the polygon. When in any other mode you cannot delete the polygon, altough it is possible to click on the button "Delete Polygons". This then gives the message from the original post and also after clicking on that button the "Delete Polygon" option for each label is not grayed-out anymore. For me that caused confusion and I thought it was a bug.

**Comment by wkentaro** [comment]
@NiklasDHahn Thanks. I could reproduce what you said. "Delete Polygons" button sometimes gets enabled even in "Create" mode. This is not right behavior, so will work fixing it.

**Comment by liyufan** [comment]
Same problem with https://github.com/wkentaro/labelme/releases/tag/v5.3.1

**Comment by hellowqc** [comment]
> 我尝试重现但无法重现，所以请让我知道重现的方法。



> @NiklasDHahn谢谢。我可以重现你所说的。即使在“创建”模式下，“删除多边形”按钮有时也会启用。这不是正确的行为，因此我们将努力解决它。

Hello, I tried to delete point annotations in the interface, but I couldn't delete them. Now, I can only delete them by selecting the corresponding point annotation in the Polygon Labels column. However, if there are many point annotations in an image, it will be difficult to select the point you want to delete

**Comment by NiklasDHahn** [comment]
@hellowqc which version are you using? I have installed '5.0.5'. In the version that I have you can use the 'Edit Polygon' function to delete any point. You need to activate it and then move you mouse over the point that you want to delete. The point will be highlighted and then you can right-click on it and choose 'Remove Selected Point'.

**Comment by sralvins** [comment]
still can not delete polygon label in ver.5.4.1

**Comment by wkentaro** [comment]
> *This was generated by AI during triage.*

Moving to **needs-triage** (appears resolved). Deleting a selected shape now removes both the shape and its Polygon Labels entry (`MainWindow.delete_selected_shapes` -> `Canvas.delete_shape` + `remove_labels`). This is covered by passing e2e tests on current `main` - `tests/e2e/shape_editing_test.py::test_delete_shape` and `tests/e2e/label_list_canvas_selection_test.py::test_delete_shape_removes_label_list_entry` (both green). Recommend closing unless a specific case still leaves the label behind after confirming the delete.
